src/devices/bus/isa/mda.cpp

```diff
@@ -965,3 +975,514 @@ MC6845_UPDATE_ROW( isa8_ec1840_0002_device::crtc_update_row )
+ * Register Address table from the manual
+ * Ericsson name          MDA mode   CGA mode  Standard name
+ *-------------------------------------------------------------------------------
+ * 6845 Address Registers 0x3b4      0x3d4     wo CRT Index reg
+ * 6845 Data Registers    0x3b5      0x3d5     wo CRT Data reg
+ * Mode Register 1        0x3b8      0x3d8     rw MDA/CGA mode reg (bit 0,1 & 4 incompatible) 
+ * Mode Register 2        0x3bf      0x3df     rw CRT/CPU page reg (incompatible w PCjr only)
+ * Status Register        0x3ba      0x3da     r  CGA/MDA status reg (incompatible)
+ *                                              w EGA/VGA feature ccontrol reg (not used by this board)
+ */
+WRITE8_MEMBER( isa8_epc_mda_device::io_write)
+{
+	LOG("%s: %04x <- %02x\n", FUNCNAME, offset, data);
+	hd6845s_device *hd6845s = subdevice<hd6845s_device>(MC6845_NAME);
+	switch( offset )
+	{
+		case 0x04:
+		  //LOGSETUP(" - HD6845S address write\n");
+			hd6845s->address_w( data );
+			break;
+		case 0x05:
+		  //LOGSETUP(" - HD6845S register write\n");
+			hd6845s->register_w( data );
+			break;
+		case 0x08: // Mode 1 reg
+			LOGMODE(" - Mode register 1 write: %02x\n", data);
+			LOGMODE("   MSB attribute: %s\n", (data & 0x20) == 0 ? "intensity" : "blink");
+			LOGMODE("   Horizontal px: %s\n", (data & 0x10) == 0 ? "320/LR" : "640/HR");
+			LOGMODE("   Video        : %s\n", (data & 0x08) == 0 ? "Disabled" : "Enabled");
+			LOGMODE("   Mode         : %s\n", (data & 0x02) == 0 ? "Text" : "Graphics");
+			LOGMODE("   Text columns : %d\n", (data & 0x01) == 0 ? 40 : 80);
+			m_mode_control = data;
+			m_vmode &= ~(VM_GRAPH | VM_COLS80 | VM_HOR640);
+			m_vmode |= ((m_mode_control & 0x01) ? VM_COLS80 : 0);
+			m_vmode |= ((m_mode_control & 0x02) ? VM_GRAPH  : 0);
+			m_vmode |= ((m_mode_control & 0x10) ? VM_HOR640 : 0);
+			m_update_row_type = ((data & 0x20) == 0 ? MDA_LOWRES_TEXT_INTEN : MDA_LOWRES_TEXT_BLINK);
+			{
+				rectangle rect(0, get_xres() - 1, 0, get_yres() -1);
+				m_screen->configure(get_xres(), get_yres(), rect, HZ_TO_ATTOSECONDS(50));
+			}
+			LOGMODE("Video Mode:%02x\n\n", m_vmode);
+			break;
+		case 0x0f: // Mode 2 reg
+			LOGMODE(" - Mode register 2 write: %02x\n", data);
+			LOGMODE("   Vertical px  : %s\n", (data & MR2_VER400) == 0 ? "200" : "400");
+			LOGMODE("   Character set: %s\n", (data & MR2_CHRSET) == 0 ? "0" : "1");
+			LOGMODE("   Emulated     : %s\n", (data & MR2_COLEMU) == 0 ? "Color" : "Monochrome");
+			m_mode_control2 = data;
+			m_vmode &= ~(VM_MONO | VM_VER400);
+			m_vmode |= ((m_mode_control2 & 0x04) ? VM_MONO   : 0);
+			m_vmode |= ((m_mode_control2 & 0x80) ? VM_VER400 : 0);
+			{
+				rectangle rect(0, get_xres() - 1, 0, get_yres() -1);
+				m_screen->configure(get_xres(), get_yres(), rect, HZ_TO_ATTOSECONDS(50));
+			}
+			LOGMODE("Video Mode:%02x\n\n", m_vmode);
+			break;
+		default:
+			LOG("EPC MDA: io_write at wrong offset:%02x\n", offset);
+			logerror("EPC MDA: io_write at wrong offset:%02x\n", offset);
```

Comment:
You're going to unconditionally log here and log conditionally – please don't.

Reply:
fixed